### PR TITLE
Transcripts: update shelf icon

### PR DESCRIPTION
**Pull request by leandroalonso**
| 📘 Part of: #1848 |
|:---:|

Updates transcript icon, handle being available/unavailable on the bottom sheet, handle adding it on the correct position

## Screenshots
| Shelf | Overflow | Editing |
|:---:|:---:|:---:|
|Simulator Screenshot - iPhone 15 Pro Max - 2024-08-07 at 14 54 56|Simulator Screenshot - iPhone 15 Pro Max - 2024-08-07 at 14 55 22|Simulator Screenshot - iPhone 15 Pro Max - 2024-08-07 at 14 55 42|

## Shelf without customizations

1. Run a clean install of the app
2. Play [this episode](https://pca.st/kxlbcyhl) (no transcript available) and open the player
3. ✅ There should not be any transcript icon on the shelf, tap "..." on the shelf, there should not be any mention to transcript there
4. Dismiss the player
5. Go to Profile > Settings > Beta Features > enable **transcripts**
6. Go back to the player
7. ✅ You should see the transcript as the 4th icon
8. Tap "..." and then "Edit"
9. ✅ Transcript should appear with the same colors/opacity as the rest of the items
10. Move it below
11. Tap Done
12. Tap "..." on the shelf again
13. ✅ Transcript should be there and it should be disabled, tap on it and nothing should happen

## Shelf with customizations

1. Run a clean install of the app
2. Play an episode with transcripts available (like Cautionary Tales) and open the player
3. ✅ There should not be any transcript icon on the shelf, tap "..." on the shelf, there should not be any mention to transcript there
4. Tap Edit
5. Rearrange the items and tap "Done"
6. Dismiss the player
7. Go to Profile > Settings > Beta Features > enable **transcripts**
6. Go back to the player
7. ✅ You should see transcript as the 4th icon
8. Tap "..." and then "Edit"
9. Move Transcript to below
10. Tap "Done"
11. Tap "..." on the shelf
12. ✅ Transcript should be available, tapping on it should display the transcript

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/ShelfActionsViewController+Table.swift [hunk 1]
```diff
@@ -51,11 +51,19 @@ extension ShelfActionsViewController: UITableViewDelegate, UITableViewDataSource
             } else {
                 cell.actionIcon.tintColor = ThemeColor.playerContrast02()
             }
+
+            // Disable transcript if not available
+            let isTranscriptsAndIsDisable = action == .transcript && !isTranscriptEnabled
```
Comment (leandroalonso) [inline podcasts/ShelfActionsViewController+Table.swift:56]: Not particularly proud of this. My first try was using a property on the `enum`, but with the async nature of checking transcript availability I couldn't make it work.

I extracted the logic to check for transcripts to a protocol, so there's no code duplication.

Given we will eventually remove the button from the shelf I think this is a ok trade-off for now.

File: podcasts/ShelfActionsViewController.swift [hunk 2]
```diff
@@ -148,6 +157,14 @@ class ShelfActionsViewController: UIViewController {
         actionsTable.reloadData()
     }
 
+    @objc private func episodeTranscriptAvailabilityChanged(notification: NSNotification) {
+        guard let episodeUuid = notification.userInfo?["episodeUuid"] as? String,
+              let isAvailable = notification.userInfo?["isAvailable"] as? Bool,
```
Comment (danielebogo) [inline podcasts/ShelfActionsViewController.swift:162]: Not related to this PR: I'm wondering if we can make these keys global as we re-use them in other places 🤔

**Review by danielebogo — APPROVED** [review]
Tested both cases from a fresh install and works as expected!

**Pull request merged**